### PR TITLE
Histogram aggregator functional options

### PR DESCRIPTION
**Pull request by jmacd**
Introduces a functional option `WithExplicitBoundaries()` where there was formerly an explicit `[]float64` argument. 

This is to prepare for https://github.com/open-telemetry/opentelemetry-specification/issues/982, adding multiple histogram boundary schemes.

This uses the Prometheus default histogram buckets (11) as the default for floating point histograms (0.005 ... 10). Integer histograms use the same bucket defaults multiplied by 1e6 (5_000 ... 10_000_000).

### CODE REVIEW

**Review by evantorrie — APPROVED** [review]
LGTM

**Comment by jmacd** [comment]
@bogdandrutu Please review.

File: sdk/metric/aggregator/histogram/histogram.go [hunk 1]
```diff
@@ -41,6 +41,19 @@ type (
 		state      state
 	}
 
+	// Config describes how the histogram is aggregated.
+	Config struct {
+		// ExplicitBoundaries support arbitrary bucketing schemes.  This
+		// is the general case.
+		ExplicitBoundaries []float64
```
Comment (bogdandrutu) [inline sdk/metric/aggregator/histogram/histogram.go:48]: No need to be exported (public).

File: sdk/metric/aggregator/histogram/histogram.go [hunk 2]
```diff
@@ -41,6 +41,19 @@ type (
...
+	}
+
+	// Option configures a histogram Config.
+	Option interface {
+		// Apply sets one or more Config fields.
+		Apply(*Config)
```
Comment (bogdandrutu) [inline sdk/metric/aggregator/histogram/histogram.go:54]: No need to be exported (public)

File: sdk/metric/aggregator/histogram/histogram.go [hunk 3]
```diff
@@ -41,6 +41,19 @@ type (
 		state      state
 	}
 
+	// Config describes how the histogram is aggregated.
+	Config struct {
```
Comment (bogdandrutu) [inline sdk/metric/aggregator/histogram/histogram.go:45]: No need to be exported (public)

**Review by Aneurysm9 — APPROVED** [review]

File: sdk/metric/aggregator/histogram/histogram.go (hunk 3 above)
Comment (jmacd) [reply sdk/metric/aggregator/histogram/histogram.go]: Our style guide is possible inconsistent on when to make the config struct, config struct fields, and apply methods exported. In this case I agree it's not a good idea to export. Done.

File: sdk/metric/aggregator/histogram/histogram.go (hunk 1 above)
Comment (jmacd) [reply sdk/metric/aggregator/histogram/histogram.go]: Done.

File: sdk/metric/aggregator/histogram/histogram.go (hunk 2 above)
Comment (jmacd) [reply sdk/metric/aggregator/histogram/histogram.go]: Done.

**Pull request merged**